src/App.js

```diff
@@ -280,16 +280,18 @@ class App extends React.Component {
   }
 
   handleIdentityCancel = () => {
+    const { identityIntent } = this.state
+
+    identityIntent.reject(new Error('Identity modification cancelled'))
     this.setState({ identityIntent: null })
   }
 
   handleIdentitySave = ({ address, label }) => {
     const { identityIntent } = this.state
     this.state.wrapper
       .modifyAddressIdentity(address, { name: label })
-      .then(() =>
-        this.setState({ identityIntent: null }, identityIntent.resolve)
-      )
+      .then(identityIntent.resolve)
+      .then(() => this.setState({ identityIntent: null }))
```

Comment:
I think it's more clear this way, but perhaps there was a reason for doing the resolution in the `setState()` callback?

Reply:
I agree